### PR TITLE
doc: edit Problem Description & Proposed Solution

### PR DESCRIPTION
**Pull request by rouson**
This commit
1. Attempts to make the feature and statement lists exhaustive.
2. Adds types, kind type parameters, and values that explicitly support parallelism.
3. Adds a list of image control statements, including those that pre-existed parallelism in Fortran.
3. Makes the use of statement syntax more uniform, e.g., switch `sync-all-stmt` to `sync all`.
4. Add a few links.

### CODE REVIEW

File: flang/docs/CoarrayFortranRuntime.md
Comment (ktras) [inline flang/docs/CoarrayFortranRuntime.md:37]: Is this line meant to be here? Please remove or fix to be complete sentences and in the correct place.

File: flang/docs/CoarrayFortranRuntime.md
Comment (ktras) [inline flang/docs/CoarrayFortranRuntime.md:40]:
```suggestion
  This design document proposes an application programming interface (API) to support the above features.  Implementations of some parts of the API exist in [Caffeine], a parallel runtime library targeting coarray Fortran compilers.  By defining a library-agnostic API, we envision facilitating the development of alternative parallel runtime libraries that support the same API.  One benefit of this approach is the ability to vary the communication substrate.  For example, Caffeine uses the [GASNet-EX] exascale networking middleware, whereas it might also be possible to develop wrappers that would support the proposed API with [OpenCoarrays], which uses the Message Passing Interface ([MPI]). A central aim of this document is to use a parallel runtime API in standard Fortran syntax, which enables us to leverage the Fortran to succinctly express various properties of the procedure interfaces, including argument attributes.  See [Rouson and Bonachea (2022)] for additional details.
```

**Review by ktras — CHANGES_REQUESTED** [review]
Thanks for this Damian! My review contains a couple inline code suggestions and comments. Please let me know if you have any questions on my feedback.

File: flang/docs/CoarrayFortranRuntime.md
Comment (rouson) [reply flang/docs/CoarrayFortranRuntime.md:37]: Good catch, @ktras.  Something got mistakenly deleted before I committed this.  I'll fix it.

Comment (rouson) [inline flang/docs/CoarrayFortranRuntime.md:37]:
```suggestion
One consequence of the statements being categorizing statements as image control will be the need to restrict code movement by optimizing compilers.
```

Comment (rouson) [reply flang/docs/CoarrayFortranRuntime.md]: @ktras commit c842a890b2d9c01bfce09669a8a61c0fc78efdfa addresses this so I'll mark this conversation as resolved.

**Comment by rouson** [comment]
All conversations have been resolved, but this PR is still marked as "1 change requested."  Could you review once more and approver and merge it if everything looks good to you?

**Review by ktras — APPROVED** [review]
LGTM

**Pull request merged**
